### PR TITLE
New parser swap mm process usage

### PR DESCRIPTION
**Pull request by rasrivas-redhat**

### CODE REVIEW

File: insights/specs/default.py
Comment (xiangce) [inline insights/specs/default.py:955]: Better get the `PID` from the Ps combiner directly, so you do not need to list the whole `"/proc"` dir and iterate the Ps again to get the "COMMAND".

File: insights/specs/default.py
Comment (xiangce) [inline insights/specs/default.py:959]: how about `readlines`, so do not need to split it in the following loop.

**Review by xiangce — CHANGES_REQUESTED** [review]
@rasrivas-redhat - As we discussed yesterday, I think we can implement this parser a bit generic, by making it supporting all the Keys contained in the `/proc/{PID}/smaps` files,  in the future, we do not need to add more spec&parsers when want to extract other keys for it..

File: insights/specs/default.py
Comment (rasrivas-redhat) [reply insights/specs/default.py]: Ack, updated the code

File: insights/specs/default.py
Comment (rasrivas-redhat) [reply insights/specs/default.py]: Ack, updated the code

File: insights/specs/default.py
Comment (xiangce) [inline insights/specs/default.py:980]: Is this `loop` essential? I think it could be done inner the above loop of `L956`

File: insights/specs/default.py
Comment (xiangce) [inline insights/specs/default.py:966]: The first part of this line looks hex number, it's incorrect to check it with `isdigit()` since the `[a-f].isdigit()` returns `False` but it's a valid hex.

Comment (xiangce) [reply insights/specs/default.py:966]: If `sum`ed these values up in the parser, some of the information will be lost.

For example, it's impossible for a rule to get the `swap` used in the specific range of the virtual memory section (eg. `7f4656856000-7f4656857000 r--p 00007000 fd:00 268516389                  /usr/lib64/libcrypt-2.17.so`) of a PID.


and the `VmFlags` should not be skiped.

File: insights/specs/default.py
Comment (xiangce) [inline insights/specs/default.py:964]: This `else` is not necessary, since you `continue` in the `if`.

File: insights/specs/default.py
Comment (xiangce) [inline insights/specs/default.py:973]: This `else continue` is not necessary too.

File: insights/specs/default.py
Comment (xiangce) [inline insights/specs/default.py:975]:
```
                       swap.update(dict(PID=int(pid), COMMAND=proc['COMMAND']))
                       smaps_data.append(swap)
```

**Review by xiangce — CHANGES_REQUESTED** [review]

**Comment by rasrivas-redhat** [comment]
@xiangce - Thanks for the feedback. As per the latest discussion with Jaylin, he is not in favor of developing this spec, he has mentioned his concerns in the comment. Apart from it, I am having a discussion with a related KCS SME about his suggestion for the best possible solution we can provide, I think tell that time we can hold on this MR.

**Comment by bfahr** [comment]
@rasrivas-redhat @xiangce I see this change is on hold, but it also will need to be changed after #3088 is merged.  I will hold off on my review until this PR is no longer on hold, and any necessary changes have been made.

**Comment by bfahr** [comment]
I've closed #3088, but added #3110 as a pattern to follow for custom datasources.  Please modify this PR to follow that pattern.

**Comment by rasrivas-redhat** [comment]
@bfahr - Thanks for sharing the information and will follow the new method whenever required. As per the discussion with the team, this spec is no more needed for the rule, so I am closing the MR.

**Pull request closed without merging**
